instructor/common.py

```diff
@@ -1309,44 +1312,85 @@ def microscopy_sample_prep(request):
     )
 
 
-def microscopy_images_edit(request, assignment, sample_prep, sp):
-    a = models.Assignment.objects.get(id=assignment)
-    sample = models.MicroscopySamplePrep.objects.get(id=sample_prep)
-    protocol = models.StrainProtocol.objects.get(id=sp)
-    MicroImagesFormset = modelformset_factory(
-        models.MicroscopyImages,
-        extra=1,
-        can_delete=True,
-        can_order=True,
-        exclude=['sample_prep', 'strain_protocol', 'order', 'image']
+def create_image_mappings(assignment, sample):
+    strain_protocols = models.StrainTreatment.objects.filter(
+        assignment=assignment
     )
-    message = ''
-    if request.method == "POST":
-        formset = MicroImagesFormset(request.POST)
-        formset.clean()
-        if formset.is_valid():
-            message = 'Thank you'
-            for form in formset.ordered_forms:
-                form.instance.order = form.cleaned_data['ORDER']
-            entries = formset.save(commit=False)
-            for form in entries:
-                form.sample_prep = sample
-                form.strain_protocol = protocol
-                form.save()
-        else:
-            message = "Something went wrong"
+    for strain_protocol in strain_protocols:
+        histogram, _ = (
+            models.MicroscopyImageMapping.objects.get_or_create(
+                sample_prep=sample,
+                strain_protocol=strain_protocol,
+            )
+        )
+
 
+@assignment_selected
+@check_assignment_owner
+@login_required
```

Comment:
I missed this in the previous review, should these three decorators also go on `microscopy_sample_prep`?